Makefile

```diff
@@ -8,268 +28,405 @@ usage:
 	@echo
 	@echo "The next step is to choose a board to build Tock for. Mainline"
 	@echo "Tock currently includes support for the following platforms:"
-	@for f in `./tools/list_boards.sh`; do printf " - $$f\n"; done
+	@for f in $(ALL_BOARDS); do printf " - $$f\n"; done
 	@echo
 	@echo "Run 'make' in a board directory to build Tock for that board,"
 	@echo "and usually 'make program' or 'make flash' to load Tock onto hardware."
 	@echo "Check out the README in your board's folder for more information."
 	@echo
 	@echo "This root Makefile has a few useful targets as well:"
-	@echo "        allboards: Compiles Tock for all supported boards"
+	@echo "        allaudit: Audit Cargo dependencies for all kernel sources"
+	@echo "       allboards: Compiles Tock for all supported boards"
 	@echo "        allcheck: Checks, but does not compile, Tock for all supported boards"
 	@echo "          alldoc: Builds Tock documentation for all boards"
-	@echo "           audit: Audit Cargo dependencies for all kernel sources"
-	@echo "              ci: Run all continuous integration tests"
 	@echo "           clean: Clean all builds"
-	@echo "          clippy: Runs the clippy code linter with Tock's default arguments"
```

Comment:
`make clippy` seems like a useful shortcut to keep around. In general, it seems useful to keep all chunks of `prepush` as easy targets, since `allcheck` takes forever.

Reply:
I propose the alternate solution: `allcheck` is now fast :)